### ISSUE TITLE
Multiple column sorts?

### ISSUE THREAD

**Issue by JeffSanchez** [bebraw/reactabular#126]
Seems like this was partially discussed / referenced in the issue regarding turning off sorting, but I just wanted to clarify: are there any near-term plans / upcoming PRs to support multi-column / complex sorts (like in the way react-datagrid supports multiple column sorts basically by default)?

If not, perhaps we'll contribute one soon...

Thanks!

**Comment by bebraw** [comment]
I don't have any concrete plans for multiple column sorting. I can see [it's quite neat](http://zippyui.com/react-datagrid/#/examples/sorting), though, and I can appreciate the idea.

Currently [SurviveJS](http://survivejs.com/) related efforts take most of time given that brings the bread to the table. My smaller projects like this are in "discuss issues and accept PRs" mode. :smile:

Given Reactabular's design, sorting logic (`src/sort_column.js`) isn't coupled with it. Now it's just provided with the distribution for convenience. I believe it would be a good idea to aim to extract that out of Reactabular and then provide instructions on how to use the logic with it. We might not need new hooks even. I think this would be a good way to solve it.

I did something similar with [react-pagify](https://github.com/bebraw/react-pagify). I was able to extract its core logic to [segmentize](https://github.com/bebraw/segmentize).

The cool thing about this kind of split is that it allows you to reuse the logic outside of React. It's a little theoretical point at the moment, but I don't see any harm in that. Extension points like this are always good and consistent with the way people like to use npm.
